### PR TITLE
Add support for relational databases

### PR DESCRIPTION
**Pull request by julianolf**
Add a new plugin called **database**. It uses [dataset](https://dataset.readthedocs.io/en/latest/index.html) which gives support for a bunch of relational databases. For more details on how to use it take a look at dataset home page.

The Redis usage by the plugins behaviours and commands were removed, now they use the database plugin, so in order to upgrade you must follow this steps:

1. On _settings.ini_ file:
1.1. Disable **irc3.plugins.storage**;
1.2. Enable **plugins.database**;
1.3 Remove the `storage = redis://...` entry;
1.4 Add an entry for the database uri like `database = sqlite:///bot.db`
2. On _alembic.ini_ file:
2.1. Set the database uri, it must match the settings file `sqlalchemy.url = sqlite:///bot.db`;
3. Update your python environment `$ pipenv update`;
4. Run the database migration `$ pipenv run alembic upgrade head`;

After this you might be ready to run the bot again.
```
$ pipenv run python mr_roboto.py settings.ini
```

Note: close issue #7

### CODE REVIEW

**Comment by bmeneguele** [comment]
I did a quick overview and seems pretty good to me. Only a minor question on the pkg version. I'm going to review once more later. But hey, really nice changes to the code overall.

File: migrations/env.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+from __future__ import with_statement
+from alembic import context
+from sqlalchemy import engine_from_config, pool
+from logging.config import fileConfig
+
+# this is the Alembic Config object, which provides
+# access to the values within the .ini file in use.
```
Comment (luizdepra) [inline migrations/env.py]: I think we should remove all this comments. They are just polluting the code.

Comment (julianolf) [reply migrations/env.py]: I agree, but I've made a few notes to create an issue just to make a clean up, adjust the code style to be in line with flake8 and normalise the comments to follow google style for documentation.

File: Pipfile.lock [hunk 2]
```diff
@@ -144,39 +166,85 @@
...
+            ],
+            "version": "==4.5.0"
+        },
+        "normality": {
+            "hashes": [
+                "sha256:11b6e06ad1d4c8edefb5dca64893777d6cbc79ae81189d52d488df01a3bfc415",
+                "sha256:739c1b40cfaefabdf537972f0284df5ce0ddb371f9475b764f478cb25bc1c4be"
             ],
-            "version": "==4.4.2"
+            "version": "==0.6.1"
```
[the remark this answers is not in the document]
Comment (julianolf) [reply Pipfile.lock]: Dunno. 🤷‍♂️ Pipenv creates the pipfile.lock. We can configure pyup to this repo and let it take care of this.

File: migrations/env.py (hunk 1 above)
Comment (bmeneguele) [reply migrations/env.py]: Thumbs up for following flake8 && google style. I'm ok with leaving these comments clean up to another PR.

File: Pipfile.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmeneguele) [reply Pipfile.lock]: Awww.. the diff totally fooled me! **"version": "==4.4.2"** is the version for **multidict** pkg! And not a downgrade from **normality**.. lol! Sorry folks.

[the remark this answers is not in the document]
Comment (luizdepra) [reply Pipfile.lock]: ahahaha

**Comment by luizdepra** [comment]
Ok, we can remove the unnecessary comments in another PR.

**Pull request merged**